processing/src/main/java/org/apache/druid/segment/filter/Filters.java

```diff
@@ -651,4 +652,31 @@ private static void generateAllCombinations(
       generateAllCombinations(result, andList.subList(1, andList.size()), nonAndList);
     }
   }
+
+  /**
+   * This method provides a "standard" implementation of {@link Filter#shouldUseIndex(BitmapIndexSelector)} which takes
+   * a {@link Filter}, a {@link BitmapIndexSelector}, and {@link FilterTuning} to determine if:
+   *  a) the filter supports bitmap indexes
+   *  b) the filter tuning specifies that it should use the index
+   *  c) the cardinality of the column is above the minimum threshold and below the maximum threshold to use the index
+   *
+   * If all these things are true, {@link org.apache.druid.segment.QueryableIndexStorageAdapter} will utilize the
+   * indexes.
+   */
+  public static boolean shouldUseIndex(
+      Filter filter,
+      BitmapIndexSelector indexSelector,
+      @Nullable FilterTuning filterTuning
+  )
+  {
+    final FilterTuning tuning = filterTuning != null ? filterTuning : FilterTuning.createDefault(filter, indexSelector);
+    if (filter.supportsBitmapIndex(indexSelector) && tuning.getUseIndex()) {
+      return filter.getRequiredColumns().stream().allMatch(column -> {
+        final int cardinality = indexSelector.getBitmapIndex(column).getCardinality();
```

Comment:
It's not obvious that `filter.supportsBitmapIndex(indexSelector)` above guarantees that `indexSelector.getBitmapIndex(column)` is non-null for every column. If it does, please add the corresponding mandate in the Javadoc for `supportsBitmapIndex()`. If it doesn't please add a null guard.

Please also annotate `BitmapIndexSelector.supportsBitmapIndex()` as `@Nullable`.

Reply:
I think of those options the one that seems nicest to me is adding a sentence like this to Filter's `getRequiredColumns`:

> If `#supportsBitmapIndex` returns true, all columns returned by this method can be expected to have a bitmap index retrievable via `BitmapIndexSelector#getBitmapIndex`.

(And/or something equivalent in `supportsBitmapIndex`)

Reply:
strengthened javadoc assertions about what is expected of these methods, also added a precondition check in `Filters.shouldUseBitmapIndex` to make sure the column has a bitmap index.